pandas/core/dtypes/cast.py

```diff
@@ -1038,6 +1038,8 @@ def maybe_cast_to_datetime(value, dtype, errors='raise'):
                                          .tz_convert(dtype.tz))
                         elif is_timedelta64:
                             value = to_timedelta(value, errors=errors)._values
+                    except OutOfBoundsDatetime:
```

Comment:
this will *also* catch the datetime.datetime case (as we have ``errors='raise'`` passed to ``to_datetime``, which will then re-raise, do we actually hit this case in testing?

Reply:
See the tests that I added, that should be covered. 

The reason that works is because this code path you are commenting on is only taken if the original data already has a np.datetime64 dtype. If you pass `maybe_cast_to_datetime` an object array of (out of bound) datetime.datetime, it will return that object array, and not raise.

Reply:
k fair enough. This code is *very* complicated, ideally like to simplify as much as possible.

Reply:
Fully agree on the code complexity .. (have another change that simplifies some of this code, but since it is unrelated to this actual bug fix, will keep that for another PR)